### PR TITLE
Use the correct Docker Image for RocksJava on Linux

### PR DESCRIPTION
**Pull request by adamretter**

### CODE REVIEW

**Comment by pdillinger** [comment]
Sorry but now the changes will have to go to the GitHub Actions config. (After merge/rebase with #12163 )  Thanks for hacking on the docker image!

**Review by ajkr — APPROVED** [review]

**Comment by adamretter** [comment]
@pdillinger I have updated this for GitHub Actions now. There are two Folly build failures, I am not certain, but I think they are unrelated to my changes, can you confirm?

File: .github/workflows/pr-jobs.yml [hunk 1]
```diff
@@ -461,19 +462,29 @@ jobs:
...
+      image: evolvedbinary/rocksjava:centos6_x64-be
       options: --shm-size=16gb
     steps:
-    - uses: actions/checkout@v4.1.0
+    # The docker image is intentionally based on an OS that has an older GLIBC version.
+    # That GLIBC is incompatibile with GitHub's actions/checkout. Thus we implement a manual checkout step.
+    - name: Checkout
+      env:
+        GH_TOKEN: ${{ github.token }}
+      run: |
```
Comment (pdillinger) [inline .github/workflows/pr-jobs.yml:473]: This is copy-pasting a complex piece of functionality. Presumably it can go into a sharable "uses" step.

Comment (adamretter) [reply .github/workflows/pr-jobs.yml:473]: @pdillinger I did try that, but unfortunately it is not possible as you end up in a "chicken or egg" situation. i.e. if you put the code in a separate GitHub action file of this repo, then it isn't yet available as the code hasn't been checked out from Git until after this step would run.

**Pull request closed without merging**